pkg/apiserver/rest/metrics.go

```diff
@@ -37,10 +38,21 @@ var DualWriterOutcome = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 	NativeHistogramBucketFactor: 1.1,
 }, []string{"mode", "name", "method"})
 
-func (m *dualWriterMetrics) init() {
+func (m *dualWriterMetrics) init(reg prometheus.Registerer) {
 	m.legacy = DualWriterLegacyDuration
 	m.storage = DualWriterStorageDuration
 	m.outcome = DualWriterOutcome
+	errLegacy := reg.Register(m.legacy)
+	errStorage := reg.Register(m.storage)
+	errOutcome := reg.Register(m.outcome)
+	if errLegacy != nil || errStorage != nil || errOutcome != nil {
+		are := &prometheus.AlreadyRegisteredError{}
+		if errors.As(errLegacy, are) || errors.As(errStorage, are) || errors.As(errOutcome, are) {
+			m.legacy = are.ExistingCollector.(*prometheus.HistogramVec)
```

Comment:
It looks like they're registered above as different collectors. In that case, wouldn't this assign the same HistogramVec to all of these?